backend/packages/Upgrade/src/loaders/typeormLoader.ts

```diff
@@ -25,53 +21,74 @@ export const typeormLoader: MicroframeworkLoader = async (settings: Microframewo
         password: env.db.password,
         database: env.db.database,
       },
-      replicas: [],
+      slaves: [],
     },
     synchronize: env.db.synchronize,
     logging: env.db.logging,
     maxQueryExecutionTime: env.db.maxQueryExecutionTime,
     entities: env.app.dirs.entities,
     migrations: env.app.dirs.migrations,
   };
-  // connection options:
-  let masterConnectionOptions = [];
-  let replicaConnectionOption = [];
-  const replicas = Object.keys(host_replicas).length;
 
+  const replicaConnectionOption = {
+    name: CONNECTION_NAME.REPLICA,
+    type: env.db.type, // See createConnection options for valid types
+    replication: {
+      master: {
+        host: '',
+        port: null,
+        username: '',
+        password: '',
+        database: '',
+      },
+      slaves: [],
+    },
+    synchronize: env.db.synchronize,
+    logging: env.db.logging,
+    maxQueryExecutionTime: env.db.maxQueryExecutionTime,
+    entities: env.app.dirs.entities,
+    migrations: env.app.dirs.migrations,
+  };
+  // number of read replicas:
+  const replicas = Object.keys(host_replicas).length;
+  const replica_hosts = [];
   if (replicas === 0) {
+    // if no read replica is defined, then we use master host for replica connection to handle export data
     /* tslint:disable:no-string-literal */
-    masterConnectionOptions = commonConnectionOptions['replication']['replicas'] = [];
-    /* tslint:disable:no-string-literal */
-    replicaConnectionOption = commonConnectionOptions['replication']['replicas'] = [];
-  } else {
-    if (replicas === 1) {
-      /* tslint:disable:no-string-literal */
-      masterConnectionOptions = commonConnectionOptions['replication']['replicas'] = [];
-    } else {
-      const replica_hosts = [];
-      for (let i = 1; i < replicas; i++) {
-        replica_hosts.push({
-          host: host_replicas[i],
-          port: env.db.port,
-          username: env.db.username,
-          password: env.db.password,
-          database: env.db.database,
-        });
-      }
-      /* tslint:disable:no-string-literal */
-      masterConnectionOptions = commonConnectionOptions['replication']['replicas'] = replica_hosts;
-    }
+    replicaConnectionOption['replication']['master'] = {
+      host: env.db.host,
+      port: env.db.port,
+      username: env.db.username,
+      password: env.db.password,
+      database: env.db.database,
+    };
+  } else if (replicas >= 1) {
+    // if a single read replica is defined, then we use the first read replica host to handle export data
     /* tslint:disable:no-string-literal */
-    replicaConnectionOption = commonConnectionOptions['replication']['replicas'] = [{
+    replicaConnectionOption['replication']['slaves'] = [{
```

Comment:
I was thinking of keeping the replica instance as read only usage, so provided it in `slaves` as slaves are used for read only operation AFAIK. We can keep it in master for replicaConnectionOption. I will make the change.

Reply:
As you can read here, the master would be used for write operation and slaves for read operation and we here require read operation in case of this replicaconnection for export data. I would say, rather we can use a simple connection for this export data without any replication as its not needed here.
<img width="1224" alt="Screenshot 2022-02-25 at 3 53 34 PM" src="https://user-images.githubusercontent.com/33730817/155698789-13bbe66f-b86a-455e-9e80-bf977c32ab56.png">

Reply:
I will take another look this morning and share what error I was seeing and how i'm testing, could easily be on my end, I will take a look at the typeorm docs also.

Reply:
Let me make sure I have the algorithm here right:

- When no replicas provided, we point `DEFAULT` and `EXPORT` connections both to the same db.
- When 1 replica is provided, we want the `EXPORT` connection to point to a different database so that we can explicitly direct that traffic there.
- When any more replicas are also provided, they will serve as read/write overflow 'slaves' for the `DEFAULT` connection

Is that the right understanding? If so, it seems like the `EXPORT` connection should not have the `replication` object, it would just be a simple connection (no master/slaves)? This is how I read the documentation, but I am not sure; the error I get is that the master connection is failing for the EXPORT connection.

Reply:
You are correct, just a minor add on in the second point:

When 1 replica is provided, we want the `EXPORT` connection to point to a read replica version of the main database(read only version of our main db instance, so that its exactly in sync with any updates made), so that we can explicitly direct the export data read operation traffic here.

As we want to use the read replica version of the main db for `EXPORT` connection I'm not sure how will this connection behave in a simple connection(no master/slaves) in case of write operation as the replica here would only be a read only one. Though as far as I know, I have attached this connection to only read requests in export data feature, so shouldn't break. If you are agreeing on this then I will make `EXPORT` connection as a simple connection.